src/frameworks/index.ts

```diff
@@ -502,13 +501,6 @@ export async function prepareFrameworks(
         },
       ]);
 
-      if (!targetNames.includes("functions")) {
```

Comment:
This included a nice optimization that we knew the codebase even if the function hadn't been deployed yet which does/should/can mean we don't actually compile all other functions' source code to fiugre out which one matches this SSR function.

Reply:
Fair, I can keep. Will pivot to the fix discussed in your branch

Reply:
Ok thought about this some more, i'll keep the ensureTargeted code and drop unshifting functions, leave that to prepare. Broken case otherwise is `--only hosting,functions`

Reply:
As we don't want to populate "only" in that case

Reply:
Ah that's correct, maybe i messed something else up when i was playing